src/liballoc/linked_list.rs

```diff
@@ -1314,6 +1316,11 @@ mod tests {
                     }
                 }
             }
+
+            // verify that the tail node points to the last node.
+            let tail = list.tail.as_ref().expect("some tail node").as_ref();
```

Comment:
Do we need the extra `as_ref` at the end here?

Reply:
`list.tail` is an `Option<Shared<Node<T>>>`. The first as_ref avoids creating a local copy (which would have to be in scope). The second converts the Shared ptr into a reference, which can then be casted.

So yes unless I'm missing something?

Reply:
Oh I see, sorry I hadn't noticed the extra level of indirection in there. Looks like the way you're doing this is about as good as it gets, since `Shared::as_ptr` returns a `*mut T`, so you'd have to cast that anyways.